Sources/Twift.swift

```diff
@@ -5,13 +5,28 @@ import Combine
 public class Twift: NSObject, ObservableObject {
   /// The type of authentication access for this Twift instance
   public private(set) var authenticationType: AuthenticationType
+  public var oauthUser: OAuth2User? {
+    switch authenticationType {
+      case .oauth2UserAuth(let user):
+        return user
+      default:
+        return nil
+    }
+  }
+  
+  /// Optional callback to be performed after a successful token refresh.
+  private var refreshCompletion: (()-> Void)? = nil
   
   internal let decoder: JSONDecoder
   internal let encoder: JSONEncoder
   
   /// Initialise an instance with the specified authentication type
-  public init(_ authenticationType: AuthenticationType) {
+  public init(_ authenticationType: AuthenticationType, onRefresh: (()-> Void)? = nil) {
     self.authenticationType = authenticationType
+      
+    if let refreshCompletion = onRefresh {
+        self.refreshCompletion = refreshCompletion
+    }
```

Comment:
We don't need to do this `if` check since the member is optional anyway!

```suggestion
    self.refreshCompletion = refreshCompletion
```